bcgov/cas-registration
# Chore: facility review fixes

There were some errors in how facility_id was being applied. 
It was being treated as a numeric where it is a UUID value, which had some cascading effects. 
This PR should fix that up. 
Also cleaned up the API a little bit by splitting out `facility-report` API/service stuff into its own directory & added some tests.

Easiest way to test this:
- Go to dashboard & create a report for Operation 3
- go to /reporting/reports/3/facilities/f486f2fb-62ed-438d-bb3e-0819b51e3aff/review (the facility ID associated with operation 3)
- There should be data in the form fields
- change some data & save
- reload the page
- see the data has changed
